cmd/coco/generate/renderTemplate.go

```diff
@@ -0,0 +1,103 @@
+package generate
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	gotemplate "text/template"
+
+	"github.com/SAP/configuration-tools-for-gitops/pkg/files"
+	"github.com/SAP/configuration-tools-for-gitops/pkg/yamlfile"
+)
+
+var (
+	filesWrite = files.Write
+)
+
+func ParseTemplate(filename string, valueFiles []string, target string) error {
+	p := parser{}
+	if err := p.parse(filename, tmplFuncs()); err != nil {
+		return fmt.Errorf("failed to parse file %q: %w", filename, err)
+	}
+
+	combinedValues, err := mergeValues(valueFiles)
+	if err != nil {
+		return err
+	}
+	var templateInputs interface{}
+	if e := combinedValues.Decode(&templateInputs); e != nil {
+		return fmt.Errorf("failed to decode values: %w", e)
+	}
+	output, err := p.execute(templateInputs)
+	if err != nil {
+		return fmt.Errorf("failed to render template %q: %w", filename, err)
+	}
+	if err := filesWrite(target, files.AllReadWrite, output); err != nil {
+		return fmt.Errorf("failed to write to file %q: %w", target, err)
+	}
+	return nil
+}
+
+type parserInt interface {
+	parse(filename string, funcs gotemplate.FuncMap) error
+	execute(data interface{}) ([]byte, error)
+}
+
+type parserMock struct {
+	Mock bool
+	Err  error
+}
+
+func (m parserMock) parse(filename string, funcs gotemplate.FuncMap) error {
+	return nil
+}
+
+func (m parserMock) execute(data interface{}) ([]byte, error) {
+	return nil, m.Err
+}
+
+type parser struct {
+	tmpl *gotemplate.Template
+}
+
+func (p *parser) parse(filename string, funcs gotemplate.FuncMap) error {
+	b, err := os.ReadFile(filename)
```

Comment:
why not call `tmplFuncs()` directly inside `parse()` since we are always passing the same function?

Reply:
addressed